zio-http/src/main/scala/zio/http/ZClient.scala

```diff
@@ -629,10 +640,11 @@ object ZClient {
       headers: Headers,
       body: Body,
       sslConfig: Option[ClientSSLConfig],
+      proxy: Option[Proxy],
     )(implicit trace: Trace): ZIO[Scope, Throwable, Response] = {
       val request = Request(version, method, url, headers, body, None)
-      val cfg     = sslConfig.fold(config)(config.ssl)
-
+      val sslCfg  = sslConfig.fold(config)(config.ssl)
+      val cfg     = sslCfg.copy(proxy = config.proxy orElse proxy)
```

Comment:
This is somewhat confusing and I think you are doing it in the wrong order. If `proxy` is specified we want to use that, otherwise we want to fall back to the configuration. I think this would be more readable as something like:

```scala
val cgf = config.copy(ssl = sslConfig.orElse(config.ssl), proxy = proxy.orElse(config.proxy))
```